source/computer/macos-setup.rst

```diff
@@ -234,15 +234,27 @@ Windows 和 Linux/macOS 系统下，`文本文件的换行符 <https://www.ruany
 
     $ brew install dos2unix unix2dos
 
-`gawk <https://www.gnu.org/software/gawk/>`__ 是用于文本处理的命令行工具。
-macOS 自带的 awk 语法上与 gawk 有所不同。推荐安装并使用 ``gawk``::
-
-    $ brew install gawk
-
 `wget <https://www.gnu.org/software/wget/>`__ 是用于下载文件的命令行工具::
 
     $ brew install wget
 
+macOS 下自带了很多实用工具，如 ``sed``、``grep`` 等（位于 :file:`/usr/bin/`\ 目录下）。
+需要注意，这些实用工具是由 BSD 提供的，而 Linux 系统下的实用工具则是由 GNU 提供的。
+BSD 和 GNU 实用工具的命令行语法有相似之处，但也有差异。二者之间的常见差异可以参考
+此\ `博文 <https://ponderthebits.com/2017/01/know-your-tools-linux-gnu-vs-mac-bsd-command-line-utilities-grep-strings-sed-and-find/>`__\ 。
+由于网络上的大部分文档介绍的都是 GNU 实用工具的用法，因而 macOS 用户在使用网络上的
+命令时可能会出现错误。这一点可以通过安装 GNU 实用工具来解决::
+
+    $ brew install autoconf binutils coreutils diffutils ed findutils flex gawk \
+        gnu-indent gnu-sed gnu-tar gnu-which gpatch grep gzip less m4 make nano \
+        screen
```

Comment:
The list is too long. From my personal experience, most of the incompatibilities come from gawk, sed, tar, find and grep. These are also the common commands we're going to mention in the "commands" section.

What about reducing the list to 

```
gawk gnu-sed gnu-tar grep findutils
```

Reply:
That's good. We may add more if someone reports more commands in the future.

Reply:
Shall we add a note so that new babies may know other commands may still exist difference and we welcome them to report to us?